data/binding/queue.go

```diff
@@ -1,20 +1,33 @@
 package binding
 
-var itemQueueIn, itemQueueOut = makeInfiniteQueue()
+import (
+	"sync"
+)
+
+var (
+	itemQueueIn      chan<- *itemData
+	itemQueueOut     <-chan *itemData
+	processOnce      sync.Once
+	queueInitialized = make(chan int)
```

Comment:
I think it would be better to use a signalling channel (the int will use up memory) https://github.com/fyne-io/fyne/blob/a25b08480d742f016ee24c1b37aa02c3bb433bd5/internal/driver/glfw/loop.go#L19 or perhaps even just a regular bool and an if condition. I think the latter would be cleaner than trying to read from a closed channel.

Reply:
> I think it would be better to use a signalling channel (the int will use up memory)

Yes, you are right, I will update :)

> or perhaps even just a regular bool and an if condition. I think the latter would be cleaner than trying to read from a closed channel.

Hmm the idea of the channel is wait for queue initialization, how could that behavior be accomplished by using just a boolean variable? 🤔

Reply:
My idea was that you would be able to send the first request within the init function, but I realized that it probably doesn't work.

Reply:
> the first request within the init function

Not sure what you mean here 🤔

Reply:
I meant that the queue only would be created once, so the channel is basically a one time use. Every other time, the channel will be closed. Thus you could make the first call to the `queueItem` run ` itemQueueIn <- &itemData{fn: f}` right after it has set up the queue and rely on a book and if outside. That was my idea anyway.

Reply:
I don't understand sorry. `queueInitialized` is there just for ensure that if` queueItem` is called in different go routines at the same time, all of them will wait until the first one that takes the `sync.Once` initialize the queue, so others don't block trying to send over a nil channel.

If you mean calling queueItem in the init package function, then I don't think that is desirable as it is going to initialize the queue at the very beginning just like before, am I missing something?

Maybe you could show me in code what you mean, I don't understand you very well 😞